src/System.Text.Utf8/src/System/Text/Utf8/Utf8CodeUnit.cs

```diff
@@ -0,0 +1,45 @@
+﻿namespace System.Text.Utf8
+{
+    public struct Utf8CodeUnit : IEquatable<Utf8CodeUnit>
```

Comment:
UTF-16's code unit is UInt16 and we called it char, I think it is more consistent to have some wrapper type (not pushing on the name). You could as well operate directly on byte[] and never create Utf8String and only extension methods to byte[]